src/services/asset-discovery-service.ts

```diff
@@ -55,15 +55,20 @@ export default class AssetDiscoveryService extends PercyClientService {
     await this.page.setJavaScriptEnabled(enableJavaScript)
 
     this.page.on('request', async (request) => {
+      if (!this.shouldRequestResolve(request)) {
+        await request.abort()
+        return
+      }
+
       if (request.url() === rootResourceUrl) {
         await request.respond({
           body: domSnapshot,
           contentType: 'text/html',
           status: 200,
         })
-      } else {
-        await request.continue()
       }
+
+      await request.continue()
```

Comment:
@blake-newman @djones I'm upgraded to `@percy/agent@0.1.10`, which includes this change, and now I get this stack trace in all my tests:
```
(node:94052) UnhandledPromiseRejectionWarning: Error: Request is already handled!
    at assert (/Users/anaulin/src/github.com/percy/percy-puppeteer/node_modules/puppeteer/lib/helper.js:278:11)
    at Request.continue (/Users/anaulin/src/github.com/percy/percy-puppeteer/node_modules/puppeteer/lib/NetworkManager.js:398:5)
    at Request.<anonymous> (/Users/anaulin/src/github.com/percy/percy-puppeteer/node_modules/puppeteer/lib/helper.js:145:23)
    at Page.page.on (/Users/anaulin/src/github.com/percy/percy-puppeteer/node_modules/@percy/agent/dist/services/asset-discovery-service.js:51:35)
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
(node:94052) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 18)
``` 

Looking that line 51 in the JS version is this line 71 in this file. Haven't investigated it in depth yet, but looks like a bug.

Reply:
Are we missing a `return` between lines 68 and 69?

Reply:
That seems very likely @anaulin.

Reply:
Fix: https://github.com/percy/percy-agent/pull/57/files